test/e2e/test/context.go

```diff
@@ -130,6 +139,17 @@ func (c Context) KubernetesMajorMinor() string {
 	return fmt.Sprintf("%d.%d", c.KubernetesVersion.Major, c.KubernetesVersion.Minor)
 }
 
+// ElasticStackVersionDefForKind returns a specific override for the given kind of resource. Defaults to an empty image
+// and the global Elastic Stack version under test if no override exists.
+func (c Context) ElasticStackVersionDefForKind(kind string) ElasticStackVersionDefinition {
```

Comment:
Is it still a _stack_ version if it's for a particular kind?

Reply:
Hm I would say so. If you want to be super precise it is a "stack component" or a "stack application" or something like that. Do you think that would be better? I think it is already a mouthful and the intent is hopefully clear?

Reply:
Ah, I was actually thinking of reducing the mouthfulness :) and going for something like `VersionForKind`. Not feeling strongly though, feel free to leave this as is.

Reply:
I used to see `GetXxxxById` hence the suggestion: `ElasticStackVersionDefByKind()`.

Reply:
I think I am going with `VersionForKind`. Thanks everyone for the suggestions, naming is hard!